php/class-admin-menu.php

```diff
@@ -43,26 +43,62 @@ public function register_menu_items() {
 			[
 				'posts_per_page' => 99,
 				'post_type'      => hello_admin()->post_type->slug,
+				'meta_key'       => 'sub_menu',
+				'orderby'        => 'meta_value_num',
+				'order'          => 'ASC',
 			]
 		);
 
 		while ( $admin_pages_query->have_posts() ) {
 			$admin_pages_query->the_post();
-			$menu_position = get_post_meta(
+			$menu_position = (int) get_post_meta(
 				get_the_ID(),
 				hello_admin()->post_type->menu_position_key,
 				true
 			);
 
-			add_menu_page(
-				get_the_title(),
-				get_the_title(),
-				'read',
-				$this->prefix . get_post_field( 'post_name' ),
-				[ $this, 'render_menu_page' ],
-				'',
-				$menu_position
+			$sub_menu = get_post_meta(
+				get_the_ID(),
+				hello_admin()->post_type->sub_menu_key,
+				true
 			);
+
+			if ( '' === $sub_menu ) {
+				update_post_meta( get_the_ID(), hello_admin()->post_type->sub_menu_key, 0 );
+			}
+
+			$title     = get_the_title();
+			$menu_slug = $this->prefix . get_post_field( 'post_name' );
+
+			if ( ! $sub_menu ) {
+				add_menu_page(
+					$title,
+					$title,
+					'read',
+					$menu_slug,
+					[ $this, 'render_menu_page' ],
+					'',
+					$menu_position
+				);
+			} else {
+				$menu_parent = get_post_meta(
+					get_the_ID(),
+					hello_admin()->post_type->menu_parent_key,
+					true
+				);
+
+				$parent_slug = $this->prefix . get_post_field( 'post_name', $menu_parent );
+
+				add_submenu_page(
+					$parent_slug,
+					$title,
+					$title,
+					'read',
+					$menu_slug,
+					[ $this, 'render_menu_page' ],
+					$menu_position
```

Comment:
Maybe we should have a second "Menu Position" setting for sub menus? At the very least we have to make sure that the parent menu item is always the first. It's possible right now to register a sub menu that appears above the parent menu in the dropdown:

<img width="164" alt="Screen Shot 2021-11-08 at 07 28 47" src="https://user-images.githubusercontent.com/1097667/140662520-c32ae6ae-0c44-48c3-a1c5-462bbb95e723.png">

Reply:
Good find! After some testing, I found that only if you set the sub menu position to 0 or less, will it be above the parent menu in the list. I was wanting to use some kind of limit or min setting in the TextControl component but those don't exist by default. Instead, I have used a conditional to set $menu_position to one if the assigned value was less than one. This way, it can never be set before the parent menu.